Address admin issues (#44)
digitalbazaar/bedrock-mongodb: README.md

bedrock.start();
```

## Configuration

For documentation on database configuration, see [config.js](./lib/config.js).

### Connecting and Authenticating
MongoDB's documentation offers tons of great examples on how to authenticate
using a myriad number of connection strings.

[Mongo Node 3.5 Driver connect docs](http://mongodb.github.io/node-mongodb-native/3.5/tutorials/connect/)

[Mongo Node 3.5 Driver atlas docs](https://docs.mongodb.com/drivers/node#connect-to-mongodb-atlas)

You can also connect to access-enabled mongo servers using some small changes to the
`config.mongodb.connectOptions`:
```js
const {config} = require('bedrock');
config.mongodb.username = 'me';
config.mongodb.password = 'password';
const {connectOptions} = config.mongodb;

> // this optional and only required if connecting to a replicaSet

Reviewer: ```suggestion
// optional, only required if connecting to a replicaSet
```

connectOptions.replicaSet = 'my_provider_replica_set';
// optional, but required in production by many providers
connectOptions.ssl = true;

> // optional but required is your provider requires tls 

Reviewer: ```suggestion
// optional, only required if your provider requires tls 
```

connectOptions.tls = true;

> // the `authSource` option replaces the older `authDB` option
> connectOptions.authSource: 'my_provider_auth_db'; 
> // an authSource should be specified else it will be the `mongodb.name`
> connectOptions.authSource = 'admin';

Reviewer: ```suggestion
// the `authSource` option replaces the older `authDB` option
// it should be specified or else it will be the `mongodb.name`
connectOptions.authSource = 'my_provider_auth_db'; 
connectOptions.authSource = 'admin';
```

> connectOptions.authSource = 'admin';

Reviewer 2: Did you intend to list `authSource` twice?

Author: good catch. I moved it down from another section and then did not notice there were 2 now. Corrected here: https://github.com/digitalbazaar/bedrock-mongodb/pull/44/commits/5a9e92e6b69a0f28a040cc3f886bf942c797d539

bedrock.start();
```

## Configuration

For documentation on database configuration, see [config.js](./lib/config.js).

### Connecting and Authenticating
MongoDB's documentation offers tons of great examples on how to authenticate

> using a myriad amount of connection strings.

Reviewer: ```suggestion
using a myriad number of connection strings.
```

[Mongo Node 3.5 Driver connect docs](http://mongodb.github.io/node-mongodb-native/3.5/tutorials/connect/)
[Mongo Node 3.5 Driver atlas docs](https://docs.mongodb.com/drivers/node#connect-to-mongodb-atlas)

You can also connect to access-enabled mongo servers using some small changes to the
`config.mongodb.connectOptions`:
```js
const {connectOptions} = bedrock.mongodb;
// this optional and only required if connecting to a replicaSet
connectOptions.replicaSet = process.env.mongo_replicaSet;

> // if you use srv in your connection string you do not need to set ssl

Reviewer: But if you don't use `srv` you do? If so...

```suggestion
// you must set `ssl` unless `srv` is present in your connection string (in which case, don't set it)
```

connectOptions.ssl = true,

> // this is new and should be user over username and password

Reviewer: ```suggestion
// `auth` should now be used instead of older options `username` and `password`
```

connectOptions.auth = { user: process.env.mongo_user, password: process.env.mongo_password },

> // this was previously call authDB

Reviewer: ```suggestion
// the `authSource` option replaces the older `authDB` option
```

> connectOptions.auth = { user: process.env.mongo_user, password: process.env.mongo_password },

Reviewer 2: with latest revision developers should not need to interact with `auth` at all correct?  If so I suggest that this be removed and replaced with verbiage about using `config.mongodb.username` and config.mongodb.password`

Author: I revised the README twice. Are we ready to go here?
